src/io_proxy.c

```diff
@@ -145,6 +145,38 @@ io_proxy_read_private_key_file(const char* pkey_file_path,
 	return 0;
 }
 
+int
+io_proxy_read_private_key(char* pkey_data,
```

Comment:
I am still confused as to why reading a private_key is handled by the IO proxy

Reply:
The names are confusing. I followed the existing pattern for reading the encryption key. The value is used by the io_proxy but the functions do not use io_proxies. I think the pattern for the io_proxy file is to prefix everything with that name. 

I'll look into moving them into the utils file.

Reply:
moved them to utils and renamed them.